pkg/gardenlet/controller/shoot/shoot/cleaner.go

```diff
@@ -184,6 +184,11 @@ func (c *cleaner) finalizeShootManagedResources(ctx context.Context, namespace s
 
 	shootMRList := &resourcesv1alpha1.ManagedResourceList{}
 	for _, mr := range mrList.Items {
+		// TODO(rfranzke): Uncomment the next line after v1.90 has been released
```

Comment:
If access secrets in the garden runtime cluster are the only problem, we might not need to revert the shoot-specific parts, correct?

Reply:
> I don't see the class property of the GRM to be configurable in its component config. It is added always ...

I see, then it makes sense to revert the entire commit.

Reply:
> I would appreciate if you create a dedicated follow-up PR to permanently resolve the TODOs.

What do you mean?
The permanent resolution is exactly the commit you are reverting here.
I understand that some external components have not been adapted to the breaking change as they were supposed to. So giving folks more time to adapt is fine. But the permanent resolution is to introduce @rfranzke's commit again.

Reply:
To be fair: there was no release note about this change, neither in https://github.com/gardener/gardener/pull/8032/ nor in https://github.com/gardener/gardener/pull/8883.

We should probably add a release note to this PR telling other components to explicitly set the `class=shoot` label for shoot access secrets and access secrets in the garden runtime cluster.
Otherwise, we might face the same situation again in a few months 😄

Reply:
Requiring access Secrets for shoot to be labelled with `resources.gardener.cloud/class=shoot` is ok. It doesn't seem ok to require access Secrets in the runtime cluster to be labelled with `resources.gardener.cloud/class=shoot` so that the token requestor controller considers. The corresponding secrets are for accessing the virtual cluster, not a Shoot cluster. The corresponding class filtering shouldn't be rather enabled for the virtual-GRM. Otherwise we ask folks to label their access Secrets with a label that does not make sense IMO.